### PR TITLE
Create slide with worker

### PR DESCRIPTION
**Pull request by crashGoBoom**
# Description

This PR adds a new function to save the newly created report to the DB for the newly registered user.

## Discussion list or pending tasks

* [x] Save file to DB for the new user.
* [x] Generate report input file from survey.

## How to test

To test create a new job using a buffalo task:
`docker-compose run havenapi buffalo task files:create_slide user1@havengrc.com`

This will trigger the create slide task for the user provided.

### CODE REVIEW

File: worker/main.go [hunk 1]
```diff
@@ -153,6 +213,94 @@ func CreateSlide(ctx worker.Context, args ...interface{}) error {
...
+			surveyData[i].Points)
+	}
+	// Supports only 1 respondent currently.
+	respondent := "1"
+	for i := range questions {
+		fileContents += fmt.Sprintf("%d,%s,%s\n", i, strings.Trim(strings.Join(strings.Fields(fmt.Sprint(questions[i])), ","), "[]"), respondent)
+	}
+
+	// Create new file
+	file, err := os.Create("/tmp/" + surveyID + ".csv")
```
Comment (statik) [inline worker/main.go]: I haven't fully thought through the scenario, but I think using predictable filenames in /tmp has some security risks: https://www.owasp.org/index.php/Insecure_Temporary_File

I also noticed that it doesn't seem like the file is removed after being used.

What do you think about using something like https://golang.org/pkg/io/ioutil/#TempFile and deleting the temporary file(s) once the job completes?

**Comment by statik** [comment]
Awesome work, and thank you for providing the info on how to activate this code in the review notes: `docker-compose run havenapi buffalo task files:create_slide user1@havengrc.com`
Could you note the testing command in the readme or somewhere in docs?

**Comment by crashGoBoom** [comment]
> Awesome work, and thank you for providing the info on how to activate this code in the review notes: `docker-compose run havenapi buffalo task files:create_slide user1@havengrc.com`
> Could you note the testing command in the readme or somewhere in docs?

So the example here in the review notes is actually an older one. We added the survey ID as an argument that is passed in and the update command is in the readme as
`docker-compose run havenapi buffalo task files:create_slide $SURVEYID $USEREMAIL`

File: worker/main.go (hunk 1 above)
Comment (crashGoBoom) [reply worker/main.go]: @statik I have refactored how these temp files are handled. It included moving some things around but let me know what you think. Thanks!

**Pull request merged**